tests/unit/test_vecvlagrange1.py

```diff
@@ -20,28 +20,32 @@ def setup(self, sd):
         div = discr.assemble_div_matrix(sd)
         symgrad = discr.assemble_symgrad_matrix(sd)
 
-        div.eliminate_zeros()
-        symgrad.eliminate_zeros()
-
         div_div = discr.assemble_div_div_matrix(sd)
         symgrad_symgrad = discr.assemble_symgrad_symgrad_matrix(sd)
 
-        div_div.eliminate_zeros()
-        symgrad_symgrad.eliminate_zeros()
-
         pen = discr.assemble_penalisation_matrix(sd)
         pen.data[np.abs(pen.data) < 1e-10] = 0
-        pen.eliminate_zeros()
 
         diff = discr.assemble_diff_matrix(sd)
-        diff.eliminate_zeros()
 
         stiff = discr.assemble_stiff_matrix(sd)
         stiff.data[np.abs(stiff.data) < 1e-10] = 0
-        stiff.eliminate_zeros()
+
+        for X in [div, symgrad, div_div, pen, symgrad_symgrad, diff, stiff]:
+            self.make_canonical(X)
 
         return M, div, symgrad, div_div, symgrad_symgrad, pen, diff, stiff, discr
 
+    def make_canonical(self, A: sps.sparray):
+        A.sum_duplicates()
```

Comment:
do you think we need to do the sum_duplicates also for the other functions in the discretization classes?

Reply:
I don't know. It seems to be only a problem here and it is not clear to me why. I would suggest we only do it if/when it becomes necessary.

We may need to look for a different way to compare sparse arrays